lib/xandra/connection.ex

```diff
@@ -546,12 +571,15 @@ defmodule Xandra.Connection do
     end
   end
 
+  def connected({:call, from}, {:checkout_state_for_next_request, _}, %__MODULE__{
+        in_flight_requests: in_flight_requests
+      })
+      when map_size(in_flight_requests) == @max_concurrent_requests do
+    {:keep_state_and_data, {:reply, from, {:error, :too_many_concurrent_connections}}}
```

Comment:
This is the semantically-correct error message, right?

```suggestion
    {:keep_state_and_data, {:reply, from, {:error, :too_many_concurrent_requests}}}
```